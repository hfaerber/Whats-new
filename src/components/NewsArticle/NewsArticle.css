* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

.newsArticle {
  color: black;
  width: 48%;
  margin: 0 2% 2% 0;
  background-color: whitesmoke;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

img {
  width: 100%;
  height: 200px;
  object-fit: cover;
}

h3,
p {
  margin: 10px;
}

.linkBox {
  padding: 10px;
  background-color: rebeccapurple;
  display: block;
  width: 100%;
  border: 0;
  font-size: 1.2em;
}

a {
  color: whitesmoke;
  text-decoration: none;
}
