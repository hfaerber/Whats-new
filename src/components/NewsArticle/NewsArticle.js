import React from 'react';
import './NewsArticle.css';
import PropTypes from 'prop-types';

function NewsArticle(props) {
  return (
    <div className="newsArticle">
      <img src={props.articleImg} alt="stock"/>
      <h3>{props.articleHeadline}</h3>
      <p>{props.articleDescription}</p>
      <a className="linkBox" href={props.articleUrl}> Link To article</a>
    </div>
)}

NewsArticle.propTypes = {
  articleImg: PropTypes.string,
  articleHeadline: PropTypes.string,
  articleDescription: PropTypes.string,
  articleUrl: PropTypes.string,
}

export default NewsArticle;
