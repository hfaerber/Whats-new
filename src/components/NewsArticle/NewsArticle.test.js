import React from 'react';
import NewsArticle from './NewsArticle';
import { shallow } from 'enzyme';

describe('NewsArticle', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<NewsArticle
      key = {1}
      articleImg = 'https://theknow.denverpost.com/wp-content/uploads/2016/03/CDXXTHEWHODSC_4082x.jpg'
      articleHeadline = 'The Who postpones Denver Concert at the Pepsi Center'
      articleDescription = 'Classic-rock band The Who has postponed'
      articleUrl = 'https://theknow.denverpost.com/2019/09/27/the-who-postpones-denver-concert/225182/'
    />);
    expect(wrapper).toMatchSnapshot();
  });
})
