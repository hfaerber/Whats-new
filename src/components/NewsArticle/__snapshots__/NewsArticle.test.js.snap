// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match the snapshot with all data passed in correctly 1`] = `
<div
  className="newsArticle"
>
  <img
    alt="stock"
    src="https://theknow.denverpost.com/wp-content/uploads/2016/03/CDXXTHEWHODSC_4082x.jpg"
  />
  <h3>
    The Who postpones Denver Concert at the Pepsi Center
  </h3>
  <p>
    Classic-rock band The Who has postponed
  </p>
  <a
    className="linkBox"
    href="https://theknow.denverpost.com/2019/09/27/the-who-postpones-denver-concert/225182/"
  >
     Link To article
  </a>
</div>
`;
